includes/Classifai/Admin/BulkActions.php

```diff
@@ -80,24 +99,75 @@ public function bulk_action_handler( $redirect_to, $doaction, $post_ids ) {
 	}
 
 	/**
-	 * Display an admin notice after classifying posts in bulk.
+	 * Handle media bulk actions.
+	 *
+	 * @param string $redirect_to       Redirect URL after bulk actions.
+	 * @param string $doaction          Action ID.
+	 * @param array  $attachment_ids    Attachment ids to apply bulk actions to.
+	 *
+	 * @return string
+	 */
+	public function media_bulk_action_handler( $redirect_to, $doaction, $attachment_ids ) {
+		if ( empty( $attachment_ids ) ) {
+			return $redirect_to;
+		}
+		switch ( $doaction ) {
+			case 'alt_tags':
+				$action = 'alt_tagged';
+				break;
+			case 'image_tags':
+				$action = 'image_tagged';
+				break;
+			case 'smart_crop':
+				$action = 'smart_cropped';
+				break;
+			default:
+				return $redirect_to;
+		}
+		foreach ( $attachment_ids as $attachment_id ) {
+			$this->save_post_handler->classify( $attachment_id );
+		}
+		$redirect_to = add_query_arg( "bulk_{$action}", count( $attachment_ids ), $redirect_to );
+		return $redirect_to;
+	}
+
+	/**
+	 * Display an admin notice after bulk updates.
 	 */
 	public function bulk_action_admin_notice() {
-		if ( empty( $_REQUEST['bulk_classified'] ) ) {
+		if ( empty( $_REQUEST['bulk_classified'] ) && empty( $_REQUEST['bulk_alt_tagged'] ) && empty( $_REQUEST['bulk_image_tagged'] ) && empty( $_REQUEST['bulk_smart_cropped'] ) ) {
 			return;
 		}
 
-		$classified_posts_count = intval( $_REQUEST['bulk_classified'] );
+		if ( ! empty( $_REQUEST['bulk_classified'] ) ) {
+			$classified_posts_count = intval( $_REQUEST['bulk_classified'] );
+			$post_type              = 'post';
+			$action                 = 'Classified';
```

Comment:
New strings will need to be translatable.

I think these will need to be full strings ( `Classified X posts`, `Alt tags generated for X images` ) to allow maximum flexibility for translations: I learnt recently that even allowing for full reordering can be problematic in some languages as the verbs nouns and number may be blended together. 

:1234: Applies to each item in this section.